core/base/eigenField/CMakeLists.txt

```diff
@@ -3,23 +3,16 @@ ttk_add_base_library(eigenField
     EigenField.cpp
   HEADERS
     EigenField.h
-  LINK
+  DEPENDS
     geometry
     laplacian
     triangulation
+  OPTIONAL_DEPENDS
+    Eigen3::Eigen
     )
 
 if (EIGEN3_FOUND)
-  find_path(SPECTRA_INCLUDE_DIR Spectra/SymEigsSolver.h)
-  if(SPECTRA_INCLUDE_DIR STREQUAL "SPECTRA_INCLUDE_DIR-NOTFOUND")
-    option(TTK_ENABLE_SPECTRA "Enable Spectra support" OFF)
-    message(STATUS "Spectra not found, disabling Spectra support in TTK.")
-  else()
-    option(TTK_ENABLE_SPECTRA "Enable Spectra support" ON)
-  endif()
-else()
-  option(TTK_ENABLE_SPECTRA "Enable Spectra support" OFF)
-  message(STATUS "Spectra not found, disabling Spectra support in TTK.")
```

Comment:
I'm confused here. where did all that code block go?

Reply:
In config.cmake as we only look for SPECTRA if NUMPY is enabled.
The idea is to link SPECTRA only for module, but to `find` it (so enable/disable it) only once.

Reply:
Sorry, my comment is indeed wrong. In `config.cmake` we only look for SPECTRA if EIGEN is enabled (not numpy, sorry).
As you can see in `config.cmake`, SPECTRA is still enabled only if EIGEN is found. I just wanted to avoid any `find_package` in a base cmake file. I this all dependencies should be resolved once in the `config.cmake`.

Reply:
gotcha. thanks!